sips/sip-030/sip-030-wallet-interface.md

```diff
@@ -0,0 +1,527 @@
+Importantly, the decision to use an existing standard (rather than designing a new one or reworking Auth) is intentional — to avoid further division or split ownership within the community.
+
+There was an attempt to re-use existing standards/protocols from other ecosystems via the WBIPs working group — but no consensus was found that was a perfect fit or had enough traction for the larger layer-2 ecosystem.
+So this SIP aims to capture the important features for the Stacks ecosystem, with a focus on extensibility.
+
+# Specification
+
+The proposed changes are listed as follows:
+
+Specify [JSON-RPC 2.0](https://www.jsonrpc.org/specification) compatible methods and payloads for wallet interaction.
+These can be used via a browser object (i.e., via the `window.btc.request` method) or similar interfaces like WalletConnect.
+
+# Backwards Compatibility
+
+The implementation of this proposal is not necessarily backward compatible.
+Wallets implementing the new standard may maintain the previous system to support legacy applications during a transition period or indefinitely.
+Existing applications using the current Auth system should continue to operate, but immediate changes are recommended once this SIP is ratified.
+
+# Implementation
+
+## Notes on Serialization
+
+To adhere to a generic serializability, the following notes are given:
+
+- Enums are serialized as human-readable strings.
+- BigInts are serialized as numbers, strings, or anything that can be parsed by the JavaScript BigInt constructor.
+- Bytes are serialized as hex-encoded strings (without a 0x prefix).
+- Predefined formats from previous SIPs are used where applicable.
+- Addresses are serialized as Stacks c32-encoded strings.
+- Clarity values, post-conditions, and transactions are serialized to bytes (defined by SIP-005) and used as hex-encoded strings.
+
+## Methods
+
+This section defines the available methods, their parameters, and result structure.
+Parameters should be considered only as recommendations for the wallet, and the user/wallet may choose to ignore or override them.
+
+> Note: Optional params are marked with a `?`.
+
+Methods can be namespaced under `stx_` if used in more generic settings and other more Ethereum inspired domains.
+In other cases (e.g. `WalletConnect`), the namespace may already be given by metadata (e.g. a `chainId` field) and can be omitted.
+On the predominant `StacksProvider` global object, the methods can be used without a namespace, but wallets may add namespaced aliases for convenience.
+
+#### Transaction method general params
+
+The following definitions can be used in the transaction methods.
+
+`params`
+
+- `address?`: `string` address, Stacks c32-encoded, defaults to wallets current address
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'mocknet'`
+- `fee?`: `number | string` BigInt constructor compatible value
+- `nonce?`: `number | string` BigInt constructor compatible value
+- `attachment?`: `string` hex-encoded
```

Comment:
Are attachments still a thing? We moved on to StackerDBs

Reply:
I've been considering opening a (separate) SIP to deprecate the Atlas subsystem now that nothing is using it.

In that vein, it may be more appropriate to call this field something non-Atlas-specific, like `content-hash`.  Then, it could be used by clients to authenticate off-chain content obtained from StackerDB or somesuch system.

Reply:
If it's not super clear when and how the content hashes would be used going forward, I'll err on the side of removing it completely (for now) — wallets can still decide to add it later